LEAR-1507 | Implement new design & POST API integration

Jbholae opened the pull request:
[JIRA](https://readytowork.atlassian.net/browse/LEAR-1507)

InProgress positioning and flow of footer, detail fetching and comment api

Jbholae committed:
chore : added axios & react-query

Jbholae committed:
chore : implement queryclient

Jbholae committed:
feat : cerated API service with axios

Jbholae committed:
feat : service for movie/movieDetail api

Jbholae committed:
feat : implemented react query

Jbholae committed:
feat : added url for comment & posts

Jbholae committed:
feat : route for tabs and socials

Jbholae committed:
chore : api for posts & comments

Jbholae committed:
feat : new screen of tabs and socials

Jbholae committed:
feat : added formik and yup

Jbholae committed:
feat : createPost

Jbholae committed:
fix : spin while loading

Jbholae committed:
feat : submit api, create post modal css

Jbholae committed:
WIP : tab for socail and movies

Jbholae committed:
feat : Created loader component

Jbholae committed:
feat : created footerComponent

Jbholae committed:
chore : created interface

Jbholae committed:
fix : added loader,
fixed navigation

Jbholae committed:
feat : added footerCompoonent

Jbholae committed:
*Reverted back to route setup

Jbholae committed:
*created screen for tab (testing purpose)

Jbholae committed:
fix : added loader

Jbholae committed:
fix : api post and set value fixed

SagarTS reviewed (commented):
Please have a look at the comments below
1. Stick footer at the bottom of the screen (visible all the time)
2. Remove unnecessary partition of tha page (grey gaps in pages)
3. Change background to suitable image or color
4. In Social, after posting, show the posted data in UI
5. Fix social get api call on every time screen is focused
6. Fix overall design.

Jbholae committed:
removed tab screen

Jbholae committed:
fix : update and fixed UI components,
created use state for posid, modals, toastmessages,formdetail values,
implemented update post value api,
managed and created Ui for detail post, created post

Jbholae committed:
fix : fixing css for empty container,
color fixing

Jbholae committed:
feat : created api for update post detail

Jbholae committed:
feat : created footer-wrapper class for sticky nav bar,
changed background color

Jbholae committed:
fix : removed unwanted css

Jbholae committed:
fix : call postDetail api only if postId is not 0,
placement for set toast message,
dynamic userId,

Jbholae committed:
fix : alignment of loader

Jbholae committed:
WIP : height to occupy full screen

Jbholae committed:
feat : profile screen,
Photo crop,
Upload photo

SagarTS reviewed (changes requested):
<img width="412" alt="Screenshot 2024-01-08 at 10 08 13 AM" src="https://github.com/Jbholae/ionic-Movie/assets/31314543/386708c2-03df-4950-852c-03e9ede396a8">

Fix issues mentioned in images, also show uploaded image only when image is cropped.

Jbholae committed:
fix : add icon for image

Jbholae committed:
fix : width of footer

Jbholae committed:
fix : displaying cropped image

SagarTS reviewed (commented):
<img width="412" alt="Screenshot 2024-01-08 at 11 21 03 AM" src="https://github.com/Jbholae/ionic-Movie/assets/31314543/303cf2c4-03eb-4aaa-83ab-a8d5bcfb9962">

Let's remove the black square that is seen once photo is cropped.

Jbholae committed:
fix : removed black square after image crop

SagarTS reviewed (approved):
LGTM! 🙇
